django/forms/models.py

```diff
@@ -1112,7 +1112,7 @@ def __iter__(self):
             yield self.choice(obj)
 
     def __len__(self):
-        return (len(self.queryset) + (1 if self.field.empty_label is not None else 0))
+        return (len(self.queryset.all()) + (1 if self.field.empty_label is not None else 0))
```

Comment:
I wasn't sure. `count()` adds a query to this test while the current approach does not:
https://github.com/django/django/blob/6f44f714c92d2966dca390ebd3054e5fb0bb0c80/tests/model_forms/tests.py#L2574-L2595